csvcubed/csvcubed/cli/inspect/metadataprinter.py

```diff
@@ -14,17 +16,32 @@
     CodelistsResult,
     ColsWithSuppressOutputTrueResult,
     DSDLabelURIResult,
+    DatasetURLResult,
+    QubeComponentResult,
     QubeComponentsResult,
 )
 from csvcubed.cli.inspect.metadatainputvalidator import CSVWType
 from csvcubed.cli.inspect.inspectsparqlmanager import (
+    select_codelist_dataset_url,
     select_cols_where_supress_output_is_true,
     select_csvw_catalog_metadata,
     select_csvw_dsd_dataset_label_and_dsd_def_uri,
     select_csvw_dsd_qube_components,
     select_dsd_code_list_and_cols,
+    select_qb_dataset_url,
 )
-
+from csvcubed.cli.inspect.inspectdatasetmanager import (
+    get_dataset_observations_info,
+    get_dataset_val_counts_info,
+    load_csv_to_dataframe,
+)
+from csvcubed.models.inspectdataframeresults import (
+    DatasetObservationsInfoResult,
+)
+from csvcubed.utils.csvdataset import (
+    transform_dataset_to_canonical_shape,
+)
+from csvcubed.models.csvcubedexception import InputTypeIsUnknownException, JsonldNotSupportedException
```

Comment:
Can you convert the `MetadataPrinter` class into a `@dataclass` please?

IIRC if you define your attributes inside the `__init__` function, our API docs tooling can't pick them up properly. If you use a `@dataclass` you can remove the init function entirely.